ggshield/verticals/secret/output/secret_text_output_handler.py

```diff
@@ -240,46 +200,65 @@ def leak_message_located(
+                        )
+                        leak_msg.write(formatted_line)
 
+                        detector_position = (
+                            min(detector_position[0], secret_position[0]),
+                            max(detector_position[1], secret_position[1]),
+                        )
+                    line_number += len(match.lines_with_secret) - 1
                 detector_position = int(detector_position[0]), int(detector_position[1])
-                detector = format_detector(flat_match.match_type, *detector_position)
+                detector = format_detector(match.match_type, *detector_position)
                 leak_msg.write(display_detector(detector, offset))
 
-        # The current line is just here for context
-        else:
-            leak_msg.write(line.build_line_count(padding, is_secret=False))
-            if clip_long_lines:
-                line_content = clip_long_line(line_content, max_width, after=True)
-            leak_msg.write(f"{display_patch(line_content)}\n")
-
-        old_line_number = multiline_end if multiline_end else line_number
+        old_line_number = line_number
 
     return leak_msg.getvalue()
 
 
 def flatten_policy_breaks_by_line(
     policy_breaks: List[PolicyBreak],
-) -> Dict[int, List[ExtendedMatch]]:
+) -> OrderedDict[Line, List[ExtendedMatch]]:
     """
     flatten_policy_breaks_by_line turns a list of policy breaks into a dictionary
     mapping a line number to a list of matches starting at that line.
     """
-    flat_match_dict: Dict[int, List[ExtendedMatch]] = dict()
+    flat_match_dict: Dict[Line, List[ExtendedMatch]] = dict()
     for policy_break in policy_breaks:
         for match in policy_break.matches:
             assert isinstance(match, ExtendedMatch)
-            span = match.span
-
-            flat_match_list = flat_match_dict.get(span.line_index_start)
-            if flat_match_list and not any(
-                x.span.column_index_start == span.column_index_start
-                for x in flat_match_list
-            ):
-                flat_match_list.append(match)
-            else:
-                flat_match_dict[span.line_index_start] = [match]
-
-    return flat_match_dict
+            for line in match.lines_before_secret + match.lines_after_secret:
+                if line not in flat_match_dict:
+                    flat_match_dict[line] = []
+            # Only add match to the first line, we will handle multiline at formating
+            line = match.lines_with_secret[0]
+            if line not in flat_match_dict:
+                flat_match_dict[line] = []
+            flat_match_dict[line].append(match)
+    # Sort the matches per line number
+    ordered_flat_match_dict: OrderedDict[Line, List[ExtendedMatch]] = OrderedDict(
+        sorted(
+            flat_match_dict.items(),
+            key=lambda item: item[0].pre_index or item[0].post_index or -1,
+        )
+    )
+    return ordered_flat_match_dict
```

Comment:
I think this can be a plain dict: `dict` preserves insertion order so when creating the dict like this, items will be correctly ordered.

Alternatively, the return value could be a `List[Tuple[Line, List[ExtendedMatch]]` since we don't need dict-like access outside of this function.